### ISSUE TITLE
Failure in Jenkins with recently released 1.23.2

### ISSUE THREAD

**Issue by IdanAdar** [michaelleeallen/mocha-junit-reporter#106]
Starting 1.23.2 after tests have passed/failed in Jenkins the following error is printed: 

```
 139 passing (16m)


=============================== Coverage summary ===============================
Statements   : 89.09% ( 1053/1182 )
Branches     : 79.29% ( 356/449 )
Functions    : 89.82% ( 150/167 )
Lines        : 89.05% ( 1041/1169 )
================================================================================
[Pipeline] junit
Recording test results
No test report files were found. Configuration error?
```

If going back to 1.23.1, everything works as expected, so looks like one of these changes created a regression: https://github.com/michaelleeallen/mocha-junit-reporter/compare/v1.23.1...master

@clayreimann You seem to be the one who pushed code most recently.
@michaelleeallen FYI.

**Comment by clayreimann** [comment]
@IdanAdar Can you provide more context?

- What version of mocha?
- What version of node?
- Are you using other software (nyc, webpack, etc.)?
- Can you provide a reproducible example project?
- Are there other errors in the output?

**Comment by jensteichert** [comment]
Having a similar issue on CircleCI with mocha 6.2.0, circleci/node:10 and nyc.

**Comment by IdanAdar** [comment]
Node 12.13, mocha 6.2.2.
This started the second 1.23.2 was released and is resolved if going back to 1.23.1.

**Comment by clayreimann** [comment]
And the problem appears to be that no result file is produced?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by holm** [comment]
We see the same on Circle CI. It seems no result file is produced.

**Comment by justinau** [comment]
Same issue here, Node 12, Mocha 7
In my case the "No test report files were found" is misleading. It's creating a result file but it's using the default filename test-results.xml, we're using --reporter-options with a custom mochaFile but it's being ignored.

fresh npm install on the repo installs mocha 5 and all tests pass, if you bump that to mocha6+ the tests fail.

**Comment by theduykh** [comment]
Facing the same issue when working with mocha-multi-reporters. 1.23.1 works well.
Please also test with mocha-multi-reporters.

**Comment by flannanl** [comment]
I have similar issue. With v1.23.1, using the options, `--reporter-options`, with a custom `mochaFile` does not work. It is always creating the results xml in the root directory with default filename, `test-results.xml`.

It seems it is related to the recent change in this line: https://github.com/michaelleeallen/mocha-junit-reporter/blob/master/index.js#L43

**Comment by walkerdb** [comment]
shot in the dark, but could it be related to mocha processing all reporter options as "reporter-option" rather than "reporter-options"? see https://github.com/mochajs/mocha/issues/4142#issuecomment-573295470

**Comment by walkerdb** [comment]
We are seeing the same issue -- this reporter stopped respecting our reporter-options on 1.23.2 . `--reporter-option mochaFile=./build/test-results-unit.xml` no longer works.

Downgrading to 1.23.1 resolved the problem for us.

**Comment by mirao** [comment]
Same issue with CodeceptJS 2.4.1, NodeJS v10.18.1, mocha@6.2.2, mocha-junit-reporter@1.23.2.
Configuration option `mochaFile: "./output/result.xml"` in CodeceptJS stopped working.
I had to downgrade to 1.23.1: `npm i mocha-junit-reporter@1.23.1`

**Comment by clayreimann** [comment]
Sorry for the heartache. I’ll see if I can–at least–get a patch release out today that reverts the changes that went in over the weekend while I look into the root cause

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by IdanAdar** [comment]
@clayreimann were you able to work out patch?

**Comment by clayreimann** [comment]
This should be resolved with 1.23.3
